### PR TITLE
8261916: gtest/GTestWrapper.java vmErrorTest.unimplemented1_vm_assert failed

### PR DESCRIPTION
**Pull request by tstuefe**
Hi,

I'd like to backport "JDK-8263728: gtest/GTestWrapper.java vmErrorTest.unimplemented1_vm_assert failed", since it prevents sporadic gtest errors.

Original commit: https://github.com/openjdk/jdk/commit/8c1112a6

This one is not a clean backport:
- `vmError.cpp` changed slightly, since https://bugs.openjdk.java.net/browse/JDK-8258479 did some code cleanups for JDK17
- more importantly, this patch touches `os::abort()`. That function had been unified into `os_posix.cpp` for all posix platforms with https://bugs.openjdk.java.net/browse/JDK-8263564 for JDK 17. I briefly considered backporting JDK-8263564, but decided against it since it is too invasive and unnecessary. Instead, I changed the various os::abort() implementations for all individual OSes, including Solaris.

Tests: Nightlies at SAP, GHAs.

Thanks, Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261916](https://bugs.openjdk.java.net/browse/JDK-8261916): gtest/GTestWrapper.java vmErrorTest.unimplemented1_vm_assert failed

### Reviewers
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/590/head:pull/590` \
`$ git checkout pull/590`

Update a local copy of the PR: \
`$ git checkout pull/590` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/590/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 590`

View PR using the GUI difftool: \
`$ git pr show -t 590`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/590.diff">https://git.openjdk.java.net/jdk11u-dev/pull/590.diff</a>

</details>

### CODE REVIEW

**Comment by tstuefe** [comment]
Windows test error in GHA unrelated.

**Comment by GoeLin** [comment]
Hi Thomas,
do you have an idea why windows pre-submit tests failed?

**Comment by tstuefe** [comment]
> Hi Thomas, do you have an idea why windows pre-submit tests failed?

Since the last cygwin update bash is crashy, which affects Windows GHAs. I see this all over the place. @shipilev is working on it. https://bugs.openjdk.java.net/browse/JDK-8276854. 

Note that the patch has been cooking for a week or so in our internal 11 test systems without problems.

**Comment by GoeLin** [comment]
Thanks for the explanation ... in the meantime I'm aware of that issue on windows, too!

**Review by GoeLin — COMMENTED** [review]
LGTM.
Yes, it is good to spread the change to the os files.  Thanks.

**Comment by tstuefe** [comment]
Thanks @GoeLin . Waiting for approval in JBS.

**Review by RealCLanger — APPROVED** [review]

**Pull request closed without merging**
